### ISSUE TITLE
Migrate from and close Stripe

### ISSUE THREAD

**Issue by benhohner** [MozillaFoundation/donate-wagtail#1372]
- [x] Research whether we can import Stripe data into braintree
- [x] contact braintree
- [ ] contact stripe to perform migration
- [ ] manually email large donors to move them over
- [ ] close stripe
- [ ] bulk email all donors to re-sign up in braintree
- [ ] create a task to remove Stripe code

**Comment by WillatMozFdn** [comment]
April 2 update from Braintree:

 Amy F. (Braintree)

Apr 2, 2021, 6:44 PM CDT
Hi Will,

This is Amy here with the Braintree Migrations team and I can help you get started with your delta data migration from Stripe. We typically go through the following steps with them:

    Stripe will dump your customer data to JSON files and use Braintree's PGP public key to encrypt the data.
    They will post these files to their SFTP server (the data files will include your customers and their payment methods). Please note that this does NOT include transactions or subscriptions.
    Once the files are on the server, Stripe will provide you with the credentials to pass along to us.
    When we have the files, we will add this migration to our schedule. As our documentation on migrations notes, the typical turnaround time is 5-10 business days.


To get started with the migration, please provide us with:

    Your Braintree merchant ID
    A confirmation from the authorized signer on your account, sent from the email address we have on file
    Please ensure that Stripe includes Network Transaction Identifiers (NTIs) when exporting your data. Stripe may refer to these values as Card Transaction IDs. These values are needed to ensure a smooth transition between payment processors.
    Please let us know if any of the stored payment methods with Stripe are Apple Pay. If so, we'll follow up with additional information on how to handle these.

We look forward to hearing back from you, and please let us know if you have any questions about this process!

I also see that you've asked if verifications can skip being sent to Kount. Currently we do not offer this as an integration option, though this functionality is currently in the works and is expected to be rolled out in the next couple months. Our documentation will be updated as soon as this option is made available.

Regards,

Amy F. | Braintree Support

**Comment by WillatMozFdn** [comment]
My clarifying question back to Amy:

"One quick question before I loop in our developer team who'll be working with me to accomplish this project.

You noted that the files that Stripe will encrypt and post to their SFTP site does not/will not include subscriptions.  What we're trying to migrate over (i.e. the primary driver of this process) is a few hundred ongoing, month recurring donors, who somehow got left behind in the initial migration about a year ago.  If Stripe doesn't include subscription information ... how would you recommend that we create recurring monthly donations in Braintree?  Is there a particular technique that your clients typically use to transfer and import their recurring monthly customers?

I would typically just lean on our previous experience doing that ... but unfortunately our lead developer who worked on the migration has moved on from the Mozilla Foundation, and we're not having much luck figuring out how he accomplished this last year."

**Comment by WillatMozFdn** [comment]
... and her response:

Hey Will,

Great question! Assuming you're using Braintree's recurring billing tool, each subscription will require its own Subscription: Create request to recreate it on our end. You may find it valuable to gather all subscription-related data beforehand so that you're ready to roll as soon as the migration is completed. You'll also want to work with your dev team to write a script to make the subscription recreation process as smooth as possible.

Fortunately, data mapping between Stripe and Braintree is fairly straightforward and we should be able to preserve any existing token ID values for your stored payment methods. This means you should already know the Payment Method Token ID values needed for each customer's subscription (we'll also send you an import log after the migration is completed; this will confirm the token IDs within the Braintree Vault).

If you already know which customers/tokens will be migrated in this batch, I would suggest pulling the next billing date associated with each customer's subscription. The next billing date can then be passed as the First Billing Date when recreating the subscription through Braintree.

The last few points I mention are based on the assumption that you'll be migrating into gateway h8yc8prjc9h3nxxd, but if this is not the correct gateway you can view any current Braintree plans by following these steps:

    Log into the Control Panel
    Click on Subscriptions in the navigation bar
    Click the Plans tab


The plans in gateway h8yc8prjc9h3nxxd are all set up to bill on a monthly basis and each has a base price of 0.01 (this is likely intended to be overridden on a case-by-case basis by passing an amount in each Subscription: Create request). The only difference between each plan seems to be the base currency. With this in mind, you'll also want to track the currency and amount being billed for each subscription on your list so that these can be kept consistent when recreating through Braintree. The amount will be passed within the Price parameter, and the currency will be determined by the value passed as the Merchant Account ID. The Plan ID must also be passed (this is a required field when creating any new subscription).

In summary, you'll need to pass all of the following parameters (at a minimum) when recreating these subscriptions through Braintree:

    Payment Method Token ID 
    First Billing Date
    Plan ID
    Price
    Merchant Account ID


You should be able to determine all of this information ahead of time (assuming you know which customers/tokens will be migrated), which should make it easier to create a list to iterate through once you're ready to actually create the subscriptions. 

Hopefully this gives you a good place to start from, and if you have any other questions please don't hesitate to ask!  

Regards,

Amy F. | Braintree Support
